src/backend/base/langflow/components/tools/mcp_sse.py

```diff
@@ -0,0 +1,101 @@
+# from langflow.field_typing import Data
+import asyncio
+from contextlib import AsyncExitStack
+
+import httpx
+from dotenv import load_dotenv
+from mcp import ClientSession
+from mcp.client.sse import sse_client
+
+from langflow.base.mcp.util import create_tool_coroutine, create_tool_func
+from langflow.components.tools.mcp_stdio import create_input_schema_from_json_schema
+from langflow.custom import Component
+from langflow.field_typing import Tool
+from langflow.io import MessageTextInput, Output
+
+load_dotenv()  # load environment variables from .env
+
+# Define constant for status code
+HTTP_TEMPORARY_REDIRECT = 307
+
+
+class MCPSseClient:
+    def __init__(self):
+        # Initialize session and client objects
+        self.write = None
+        self.sse = None
+        self.session: ClientSession | None = None
+        self.exit_stack = AsyncExitStack()
+
+    async def pre_check_redirect(self, url: str):
+        """Check if the URL responds with a 307 Redirect."""
+        async with httpx.AsyncClient(follow_redirects=False) as client:
+            response = await client.request("HEAD", url)
+            if response.status_code == HTTP_TEMPORARY_REDIRECT:
+                return response.headers.get("Location")  # Return the redirect URL
+        return url  # Return the original URL if no redirect
+
+    async def connect_to_server(
+        self, url: str, headers: dict[str, str] | None, timeout_seconds: int = 500, sse_read_timeout_seconds: int = 500
+    ):
+        if headers is None:
+            headers = {}
+        url = await self.pre_check_redirect(url)
+
+        async with asyncio.timeout(timeout_seconds):
+            sse_transport = await self.exit_stack.enter_async_context(
+                sse_client(url, headers, timeout_seconds, sse_read_timeout_seconds)
+            )
+            self.sse, self.write = sse_transport
+            self.session = await self.exit_stack.enter_async_context(ClientSession(self.sse, self.write))
+
+            await self.session.initialize()
+
+            # List available tools
+            response = await self.session.list_tools()
+            return response.tools
+
+
+class MCPSse(Component):
+    client = MCPSseClient()
+    tools = None
+    tool_names = []
+    display_name = "MCP Tools (SSE)"
+    description = "Use as a template to create your own component."
```

Comment:
Could you provide a description for the components? I don't quite understand what they are for yet